docs/parameters-schema-as-type-object.md

```diff
@@ -0,0 +1,136 @@
+# ParametersSchemaAsTypeObject
+
+## Category
+
+ARM Error
+
+## Applies to
+
+ARM OpenAPI(swagger) specs
+
+## Related ARM Guideline Code
+
+- RPC-Arg-V1-01
+
+## Output Message
+
+The schema for body parameters must specify type:object and include a definition for its reference model.
+
+## Description
+
+The schema for body parameters should be defined strictly with the type set as 'object', as ARM does not support other types of definitions. This restriction is in place to ensure a better customer experience.
+
+## How to fix the violation
+
+Please set schema "type:object in the parameters and define the reference model of the object.
+
+## Good Examples
+
+```json
+...
+{
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      // remove schema
+      "type": "string",
+      "description": "The HDInsight client API Version."
+    }
```

Comment:
highlight that it's a good example since it's a query param not a body param
```suggestion
    "ApiVersionParameter": {
      "name": "api-version",
      // not a body parameter
      "in": "query",
      "required": true,
      // no schema
      "type": "string",
      "description": "The HDInsight client API Version."
    }
```